client/lib/domains/index.js

```diff
@@ -144,10 +145,31 @@ function hasMappedDomain( domains ) {
 	return getMappedDomains( domains ).length > 0;
 }
 
+/**
+ * Parse the tld from a given domain name, semi-naively. The function
+ * first parses against a list of tlds that have been sold on WP.com
+ * and falls back to a simplistic "everything after the last dot" approach
+ * if the whitelist failed. This is ultimately not comprehensive as that
+ * is a poor base assumption (lots of second level tlds, etc). However,
+ * for our purposes, the approach should be "good enough" for a long time.
+ *
+ * @param {string}     domainName     The domain name parse the tld from
+ * @return {string}                   The TLD or an empty string
+ */
 function getTld( domainName ) {
 	const lastIndexOfDot = domainName.lastIndexOf( '.' );
 
-	return lastIndexOfDot !== -1 && domainName.substring( lastIndexOfDot + 1 );
+	if ( lastIndexOfDot === -1 ) {
+		return '';
+	}
+
+	let tld = parseDomainAgainstTldList( domainName, getWpcomTldsCachedList() );
+
+	if ( ! tld ) {
+		tld = domainName.substring( lastIndexOfDot + 1 );
+	}
+
```

Comment:
I wondered about filtering out `br` here, but I suppose it's not really any better or worse than other non-tlds.

Reply:
Eh? I'm feeling goofy but I'm not following.

Reply:
Don't worry about it.

I was just thinking about accepting illegitimate domains, and `br` seems worse somehow than some random string because `.com.br` is ok and `.br` is not. I can't think of any instance where it would actually make any difference, though, so sorry I said anything :p